main.go

```diff
@@ -16,65 +16,65 @@ const (
 
 // Training общая структура для всех тренировок
 type Training struct {
-	...        	// тип тренировки
-	...         // количество повторов(шаги, гребки при плавании)
-	...       	// длина одного шага или гребка в м
-	... 		// продолжительность тренировки
-	...       	// вес пользователя в кг
+	TrainingType string        // тип тренировки
+	Action       int           // количество повторов (шаги, гребки при плавании)
+	LenStep      float64       // длина одного шага или гребка в м
+	Duration     time.Duration // продолжительность тренировки
+	Weight       float64       // вес пользователя в кг
 }
 
 // distance возвращает дистанцию, которую преодолел пользователь.
-// Формула расчета:
-// количество_повторов * длина_шага / м_в_км
 func (t Training) distance() float64 {
-	// вставьте ваш код ниже
-	...
+	return float64(t.Action) * t.LenStep / MInKm
 }
 
-// meanSpeed возвращает среднюю скорость бега или ходьбы.
+// meanSpeed возвращает среднюю скорость тренировки.
 func (t Training) meanSpeed() float64 {
-	// вставьте ваш код ниже
-	...	
+	return t.distance() / t.Duration.Seconds() * MinInHours
```

Comment:
Здесь необходимо обработать случай, когда значение длительности равно нулю. В этом случае функция должна возвращать ноль.

Так же можно сделать так
```suggestion
	return t.distance() / t.Duration.Hours()
```